kani-compiler/src/codegen_cprover_gotoc/overrides/hooks.rs

```diff
@@ -54,104 +55,98 @@ fn matches_function(tcx: TyCtxt, instance: Instance, attr_name: &str) -> bool {
 /// <https://github.com/model-checking/kani/blob/main/rfc/src/rfcs/0003-cover-statement.md>
 /// for more details.
 struct Cover;
-impl<'tcx> GotocHook<'tcx> for Cover {
-    fn hook_applies(&self, tcx: TyCtxt<'tcx>, instance: Instance<'tcx>) -> bool {
+impl GotocHook for Cover {
+    fn hook_applies(&self, tcx: TyCtxt, instance: Instance) -> bool {
         matches_function(tcx, instance, "KaniCover")
     }
 
     fn handle(
         &self,
-        tcx: &mut GotocCtx<'tcx>,
-        _instance: Instance<'tcx>,
+        gcx: &mut GotocCtx,
+        _instance: Instance,
         mut fargs: Vec<Expr>,
-        _assign_to: Place<'tcx>,
-        target: Option<BasicBlock>,
-        span: Option<Span>,
+        _assign_to: Place,
+        target: Option<BasicBlockIdx>,
+        span: Span,
     ) -> Stmt {
         assert_eq!(fargs.len(), 2);
         let cond = fargs.remove(0).cast_to(Type::bool());
         let msg = fargs.remove(0);
-        let msg = tcx.extract_const_message(&msg).unwrap();
+        let msg = gcx.extract_const_message(&msg).unwrap();
```

Comment:
We'll have to close https://github.com/model-checking/kani/issues/45 if you keep doing that 😆